source/language/types.rst

```diff
@@ -135,8 +124,63 @@ circuits. Physical qubits must not be declared and they are, as all the qubits,
    qubit γ;
    // Declare a qubit register with 20 qubits
    qubit[20] qubit_array;
+
+.. _physical-qubits:
+
+Physical Qubits
+~~~~~~~~~~~~~~~
+
+Physical qubits refer to particular hardware qubits. Therefore, they are only fully defined with
+respect to a target device that has a published device topology. The hardware provider determines
+the integer-labels associated with each qubit within the device's topology.
+
+While virtual qubits can be named, hardware qubits are referenced by the syntax ``$[NUM]``, where
+``NUM`` is a non-negative integer. Any integer included in the published device topology is a valid
+physical qubit identifier. Note that this implies that physical qubit identifier indices may be
+non-consecutive, depending on the device.
+
+Like virtual qubits, physical qubits are global variables, but unlike virtual qubits, they must
+not be declared.
+
+These qubit types are often used for ``defcal`` blocks because calibrations are typically valid
```

Comment:
Just to clarify, defined calibrations apply only to physical qubits (cf. https://openqasm.com/language/pulses.html). The OpenQASM spec allows for a family of calibrations to be defined using a shorthand syntax, like `defcal rz(angle theta) q { ... }` but technically the `q` here is not a virtual qubit -- it's a kind of formal parameter.

> One can define a defcal using a regular identifier for a qubit, which causes that calibration definition to be valid for all physical qubits. This is most likely to be useful for gates that are implemented virtually.

Reply:
Awesome, will update.